### ISSUE TITLE
[bug]: Unable to navigate command component using Keyboard when Mac Voiceover screen reader is enabled

### ISSUE THREAD

**Issue by 1ewiswarren** [shadcn-ui/ui#8853]
### Describe the bug

This is quite a problematic bug where accessibility is concerned as navigating this component is extremely difficult. It seems to stem from the focusable element when opening the command palette.

### Affected component/components

Command

### How to reproduce

1. Enable Mac Voiceover
2. Navigate to the demo command component on the [ShadCN site](https://ui.shadcn.com/docs/components/command)
3. Try navigating the list options using a keyboard

### Logs

```bash

```

### System Info

```bash
MacOS - Chrome, Brave, Firefox, Safari
```

### Before submitting

- [x] I've made research efforts and searched the documentation
- [x] I've searched for existing issues
